client/components/drop_down/drop_down_item.jsx

```diff
@@ -0,0 +1,67 @@
+import styled, { keyframes } from 'styled-components'
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+
+export class DropDownItem extends Component {
+  static propTypes = {
+    active: PropTypes.bool,
+    index: PropTypes.number.isRequired,
+    onClick: PropTypes.func.isRequired,
+    title: PropTypes.string,
+    children: PropTypes.any
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      isActive: props.active || false
+    }
+  }
+
+  render () {
+    const {
+      active,
+      children,
+      index,
+      onClick,
+      title
+    } = this.props
+
+    return (
+      <div className='DropDownItem' key={'DropDownItem-' + index}>
+        <div
+          className='DropDownItem__title'
+          onClick={() => onClick(index)}
+          data-active={active}
+        >
+          <h1 className={!title && 'placeholder'}>
+            {title || 'Missing Title'}
+          </h1>
+          <div className='icon' />
+        </div>
+        {active &&
+          <DropDownItemContent className='DropDownItem__content'>
+            {children}
+          </DropDownItemContent>
+        }
+      </div>
+    )
+  }
+}
+
+const navKeyframes = keyframes`
+  from {
+    max-height: 0px;
+    opacity: 0;
+  }
+  to {
+    max-height: 100%;
+    opacity: 1;
+  }
+`
+const DropDownItemContent = styled.div`
+  animation-name: ${navKeyframes};
+  animation-duration: 1s;
+  animation-timing-function: ease;
+`
```

Comment:
In trying to debug the Gucci header, I found that the use of animation wasn't really necessary for these kinds of CSS movements. When you have two states -- closed + open item, it's a lot easier to manage something like `data-state=open` with `transition` instead. With this, we only get an animation on open, and not for the close. We'd have to create a second animation for the closed state. It also causes unnecessary re-renders of the items.

Reply:
Tried doing it with data-state, but still ran into some issues because elements are being added/removed from the DOM.  I think https://github.com/reactjs/react-transition-group might offer a solution, but going to save for another time.

Reply:
Yeah, what I'm proposing is to avoid adding and removing the elements from the DOM. So there's just a simple class or data toggle that shows or hides content with a transition.